packages/nextjs/utils/abi-ninja/proxyContracts.ts

```diff
@@ -89,7 +89,24 @@ export const detectProxyTarget = async (proxyAddress: Address, client: UsePublic
     throw new Error("Beacon method calls failed");
   };
 
-  const detectionMethods = [detectUsingBytecode, detectUsingEIP1967LogicSlot, detectUsingEIP1967BeaconSlot];
+  const detectUsingOpenZeppelinSlot = async () => {
+    const implementationAddr = await client.getStorageAt({
+      address: proxyAddress,
+      slot: OPEN_ZEPPELIN_IMPLEMENTATION_SLOT,
+    });
+    const resolvedAddress = readAddress(implementationAddr);
+    if (resolvedAddress === "0x" + "0".repeat(40)) {
+      throw new Error("Zero address in OpenZeppelin implementation slot");
+    }
```

Comment:
Ohh I think this check is redundant since `readAddress` already checks for zero address and throw's error, so we could remove this?